trunk/3rdparty/st-srs/stk.c

```diff
@@ -72,14 +74,34 @@ _st_stack_t *_st_stack_new(int stack_size)
             _st_num_free_stacks--;
             ts->links.next = NULL;
             ts->links.prev = NULL;
-            return ts;
+            returnt ts;
         }
     }
+#endif
+
+    extra = _st_randomize_stacks ? _ST_PAGE_SIZE : 0;
+#ifndef MD_CACHE_STACK
+    for (qp = _st_free_stacks.next; qp != &_st_free_stacks;) {
+        ts = _ST_THREAD_STACK_PTR(qp);
+        // Before qp is freed, move to next one, because the qp will be freed when free the ts.
+        qp = qp->next;
+
+        ST_REMOVE_LINK(&ts->links);
+        _st_num_free_stacks--;
+
+#if defined(DEBUG) && !defined(MD_NO_PROTECT)
+        mprotect(ts->vaddr, REDZONE, PROT_READ | PROT_WRITE);
+        mprotect(ts->stk_top + extra, REDZONE, PROT_READ | PROT_WRITE);
+#endif
+
+        _st_delete_stk_segment(ts->vaddr, ts->vaddr_size);
+        free(ts);
+    }
+#endif
```

Comment:
only rarely a few part of codes in `st-srs`, add comments to denote which `#if` to match.
```suggestion
#endif /* MD_CACHE_STACK */
```
I would suggest add comments in this way.